### PR TITLE
[L0] Return success on enable/disable p2p.

### PR DESCRIPTION
**Pull request by JackAKirk**
https://github.com/oneapi-src/unified-runtime/pull/1429 enabled the queries for p2p, but didn't correspondingly update the enable/disable functions.
This means that the portable P2P workflows, as exampled in the dpc++ tests, e.g. https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/USM/P2P/p2p_access.cpp, are broken on L0 devices.

This fixes that by implementing these functions as a no-op and returning `UR_RESULT_SUCCESS`. This is the expected behaviour since L0 enables p2p on all devices within a platform by default (see e.g. discussions here https://github.com/intel/llvm/pull/6104).

### CODE REVIEW

**Review by nrspruit — COMMENTED** [review]
LGTM, can you link your accompanying intel/llvm PR showing the e2e test results?

**Comment by pbalcer** [comment]
All CI failures are unrelated. The E2E tests that failed are known to be broken in our system (I'll add a suppress).

```
Failed Tests (9):
  SYCL :: Matrix/SG32/get_coord_int8_matB.cpp
  SYCL :: Matrix/element_wise_all_ops_1d.cpp
  SYCL :: Matrix/element_wise_all_ops_1d_cont.cpp
  SYCL :: Matrix/element_wise_all_ops_scalar.cpp
  SYCL :: Matrix/get_coord_int8_matB.cpp
  SYCL :: Matrix/joint_matrix_apply_bf16.cpp
  SYCL :: Matrix/joint_matrix_apply_two_matrices.cpp
  SYCL :: Matrix/joint_matrix_bfloat16.cpp
  SYCL :: Matrix/joint_matrix_bfloat16_array.cpp
```

**Comment by JackAKirk** [comment]
> LGTM, can you link your accompanying intel/llvm PR showing the e2e test results?

The intel/llvm CI only has a single gpu, therefore these results return early. They have to be tested locally.

**Comment by JackAKirk** [comment]
> > LGTM, can you link your accompanying intel/llvm PR showing the e2e test results?
> 
> The intel/llvm CI only has a single gpu, therefore these results return early. They have to be tested locally.

Apparently UR ci has more devices so the ur test should run, but I realized the macro strings were missing. I also forgot to add it for hip in https://github.com/oneapi-src/unified-runtime/pull/1369 so I've added the string for hip in this PR too.

**Review by nrspruit — APPROVED** [review]
LGTM for level zero

**Review by pbalcer — APPROVED** [review]

File: source/adapters/level_zero/device.cpp [hunk 1]
```diff
@@ -247,6 +247,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(
     SupportedExtensions += ("ur_exp_command_buffer ");
     // Return supported for the UR multi-device compile experimental feature
     SupportedExtensions += ("ur_exp_multi_device_compile ");
+    SupportedExtensions += ("ur_exp_usm_p2p ");
```
[the remark this answers is not in the document]
Comment (JackAKirk) [reply source/adapters/level_zero/device.cpp:250]: @oneapi-src/level-zero-maintainers Do you have an answer to the above question?

[the remark this answers is not in the document]
Comment (ldrumm) [reply source/adapters/level_zero/device.cpp:250]: We'll just merge this, then I'll post a cleanup patch to get rid of all of them

**Pull request merged**